util/topgen/merge.py

```diff
@@ -560,6 +569,25 @@ def amend_alert(top):
                 ip[0]["alert_list"]))
 
 
+def amend_wkup(topcfg: OrderedDict):
+
+    if "wakeups" not in topcfg or topcfg["wakeups"] == "":
+        topcfg["wakeups"] = []
+
+    # create list of wakeup signals
+    for m in topcfg["module"]:
+        log.info("Adding wakeup from module %s" % m["name"])
+        for entry in m["wakeup_list"]:
+            log.info("Adding singal %s" % entry["name"])
+            topcfg["wakeups"].append("{module}.{signal}".format(module=m["name"].lower(),
+                                                                signal=entry["name"]))
+
+    # add wakeup signals to pwrmgr connections
+    # TBD: What's the best way to not hardcode this signal below?
+    #      We could make this a top.hjson variable and validate it against pwrmgr hjson
```

Comment:
yes, I think that would be good since this signal name depends on what is implemented in the design. 

WDYT, @eunchan?

Reply:
It is more like chicken and egg problem. Need to sweep wakeup first to gather the list and generate PWRMGR, then PWRMGR IP data structure (in top.gen.hjson) has inter_signal_list generated. Then inter module signal will handle this correctly. In addition to that, each IP should define the wakeup signal in inter_signal_list. (Or while generating pwrgen, it can add that structure in each IP)